### PR TITLE
9567: correct solve_linear return values and capture in solve

### LINKED ISSUE #9567: solve_univariate_inequality gives wrong solution to the rational inequality abs(1/(x-1)) > 1
In current SymPy (commit c2ffd9), I see
```python
In [1]: x = symbols('x', real=True)
In [2]: from sympy.solvers.solveset import solveset
In [3]: solveset(abs(1/(x-1)) > 1)
Out[3]: (1, 2)
```
while the result should be ```(0, 1) ∪ (1, 2)```.

The following similar case works:
```python
In [4]: solveset(abs(1/x) > 1)
Out[4]: (-1, 0) ∪ (0, 1)
```
@aktech - can you please take a look?

### PR DESCRIPTION
**Pull request by smichr**
closes #9567

### CODE REVIEW

**Comment by asmeurer** [comment]
Does this fix the original issue in #9567 as well?

**Comment by asmeurer** [comment]
I don't understand what solve_linear does, but if it fixes the issue then +1.

**Comment by smichr** [comment]
I showed on the issue page that the reported issue returns correctly. 

As for solve_linear, it is trying to solve for a linear variable and also report (early in the solve process) if there is no solution possible (like when a  variable appears in an expression but actually cancels after simplification).

I can't believe we didn't see this error earlier.

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -1956,12 +1957,17 @@ def solve_linear(lhs, rhs=0, symbols=[], exclude=[]):
                                 i.function.is_number]
                         # do a slight bit of simplification
                         vi = expand_mul(vi.subs(irep))
-                        if not d.has(xi) or not (d/xi).has(xi):
+                        if xi not in dfree:
+                            return xi, vi
+                        if xi not in (d/xi).free_symbols:
```
Comment (smichr) [inline sympy/solvers/solvers.py]: think more about this...seems like this is equivalent to `xi not in dn.free_symbols` which is already the condition of this `if-block`. Perhaps both of these tests are not necessary and `return xi, vi` is all that is needed. Then factor_terms would not be needed.

[ ] check to see what (if anything) gets by these two tests.

**Comment by asmeurer** [comment]
I guess what I don't understand is why it sometimes returns a solution and sometimes returns the numerator and denominator.

**Comment by smichr** [comment]
> why it sometimes returns a solution and sometimes returns the numerator and denominator

it should probably have been a private function of solve; I think the original idea was to not throw away the extracted numerator which is needed for the more extended work of solve. Since the function was returning 2 values when it found a solution it made sense to return the denominator with the numerator, too, so 2 values would be returned when there wasn't a linear solution.

**Pull request merged**